api/v1/reviews/PKPReviewController.php

```diff
@@ -0,0 +1,225 @@
+<?php
+
+/**
+ * @file api/v1/reviews/PKPReviewController.php
+ *
+ * Copyright (c) 2023 Simon Fraser University
+ * Copyright (c) 2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPReviewController
+ *
+ * @ingroup api_v1_reviews
+ *
+ * @brief Handle API requests for reviews operations.
+ *
+ */
+
+namespace PKP\API\v1\reviews;
+
+use APP\facades\Repo;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Support\Facades\Route;
+use PKP\core\PKPApplication;
+use PKP\core\PKPBaseController;
+use PKP\core\PKPRequest;
+use PKP\db\DAORegistry;
+use PKP\log\SubmissionEmailLogEntry;
+use PKP\security\authorization\ContextAccessPolicy;
+use PKP\security\authorization\SubmissionAccessPolicy;
+use PKP\security\authorization\UserRolesRequiredPolicy;
+use PKP\security\Role;
+use PKP\submissionFile\SubmissionFile;
+
+class PKPReviewController extends PKPBaseController
+{
+    /**
+     * @copydoc \PKP\core\PKPBaseController::getHandlerPath()
+     */
+    public function getHandlerPath(): string
+    {
+        return 'reviews';
+    }
+
+    /**
+     * @copydoc \PKP\core\PKPBaseController::getRouteGroupMiddleware()
+     * @throws \Exception
+     */
+    public function getRouteGroupMiddleware(): array
+    {
+        return [
+            'has.user',
+            'has.context',
+            self::roleAuthorizer([
+                Role::ROLE_ID_SITE_ADMIN,
+                Role::ROLE_ID_MANAGER,
+                Role::ROLE_ID_REVIEWER,
+            ]),
+        ];
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function getGroupRoutes(): void
+    {
+        Route::get('history/{submissionId}/{reviewRoundId}', $this->getHistory(...))
+            ->name('review.get.submission.round.history')
+            ->whereNumber(['reviewRoundId', 'submissionId']);
+    }
+
+    /**
+     * @copydoc \PKP\core\PKPBaseController::authorize()
+     */
+    public function authorize(PKPRequest $request, array &$args, array $roleAssignments): bool
+    {
+        $illuminateRequest = $args[0]; /** @var \Illuminate\Http\Request $illuminateRequest */
+        $actionName = static::getRouteActionName($illuminateRequest);
+
+        $this->addPolicy(new UserRolesRequiredPolicy($request), true);
+        $this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
+
+        if ($actionName === 'getHistory') {
```

Comment:
`getHistory` is the only function offered by this controller, right? It might be better to just remove this test entirely and always add the policy. If a later function gets added to the handler, it'll probably need the same policy -- and as it stands, the policy wouldn't be applied.

Reply:
You are right. Removed.